sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -213,6 +215,23 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
   }
 }
 
+/**
+ * Replace the V2 data source of table in [[InsertIntoTable]] to V1 [[FileFormat]].
+ * E.g, with temporary view `t` using [[FileDataSourceV2]], inserting into  view `t` fails
+ * since there is no correspoding physical plan.
+ * This is a temporary hack for making current data source V2 work. It should be removed
+ * when catalog of data source v2 is finished.
```

Comment:
this is not about catalog. If we are inserting to a table, the `FindDataSourceTable` rule already does the fallback by calling `providingClass`. This rule only handles inserting to temp view, which needs the data source to implement write API.